packages/amazon-sumerian-hosts-core/src/core/awspack/LexFeature.js

```diff
@@ -0,0 +1,230 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+import AbstractHostFeature from 'core/AbstractHostFeature';
+import Deferred from 'core/Deferred';
+import Utils from 'core/Utils';
+import LexUtils from 'core/awspack/LexUtils';
+
+/**
+ * The AWS LexRuntime service object.
+ * @external LexRuntime
+ * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/LexRuntime.html
+ */
+
+/**
+ * Feature class for interacting with Lex bot which the response from lex bot can be used for speech or other purpose.
+ *
+ * @extends AbstractHostFeature
+ *
+ * @property {(number|undefined)} AWS_VERSION - Gets the version of AWS SDK being
+ * used. Will be undefined until [initializeService]{@link LexFeature.initializeService}
+ * has been successfully executed.
+ * @property {Object} LEX_DEFAULTS - Default values to use with calls to {@link external:LexRuntime}.
+ * @property {string} [LEX_DEFAULTS.SampleRate='16000']
+ * @property {Object} EVENTS - Built-in messages that the feature emits. When the
+ * feature is added to a {@link core/HostObject}, event names will be prefixed by the
+ * name of the feature class + '.'.
+ * @property {string} [EVENTS.response=onLexResponseEvent] - Message that is emitted after
+ * receiving lex response for the input sent
+ * @property {Object} SERVICES - AWS services that are necessary for the feature
+ * to function.
+ * @property {external:LexRuntime} SERVICES.lexRuntime - The LexRuntime service that is used
+ * to interact with lex bot. Will be undefined until [initializeService]{@link LexFeature.initializeService}
+ * has been successfully executed
+ */
+class LexFeature extends AbstractHostFeature {
```

Comment:
What's the benefit to implementing this as a host feature rather than a self-contained bit of functionality? It seems to me that none of the LexFeature functionality has any dependency on the HostObject itself; It just ends up being an object instance attached to a host but with no direct interaction with that host. If we were to implement it as an object that could be used in Babylon on its own as well as with hosts, that would be very useful to customers. Can we do that?

Reply:
Changed the LexFeature to not inherit HostFeature